shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+#include "flutter/third_party/accessibility/ax/ax_tree.h"
+#include "flutter/third_party/accessibility/ax/ax_tree_observer.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_delegate.h"
+
+#include "flutter_accessibility.h"
+
+namespace ui {
+
+//------------------------------------------------------------------------------
+/// Use this class to maintain an accessibility tree. This class consumes
+/// semantics updates from the embedder API and produces an accessibility tree
+/// in the native format.
+///
+/// To use this class, you must provide your own implementation of
+/// FlutterAccessibility and AccessibilityBridgeDelegate.
+class AccessibilityBridge : public AXTreeObserver {
+ public:
+  //------------------------------------------------------------------------------
+  /// Delegate to handle accessibility event and route accessibility action
+  /// back to the Flutter framework.
+  ///
+  /// The accessibility events are generated when accessibility tree changes.
+  /// These events must be sent to the native accessibility system through
+  /// the native API for the system to pick up the changes
+  /// (e.g. NSAccessibilityPostNotification in MacOS).
+  ///
+  /// The accessibility actions are generated by the native accessibility system
+  /// when users interacted with the assistive technologies. Those actions
+  /// needed to be sent to the Flutter framework.
+  class AccessibilityBridgeDelegate {
+   public:
+    virtual ~AccessibilityBridgeDelegate() = default;
+    //------------------------------------------------------------------------------
+    /// @brief      Handle accessibility events generated due to accessibility
+    /// tree
+    ///             changes.
+    ///
+    /// @param[in]  targeted_event      The object that contains both the
+    /// generated
+    ///                                 event and the event target.
+    /// @param[in]  bridge              The pointer to the accessibility bridge
+    /// that
+    ///                                 can be used for querying the
+    ///                                 accessibility information at the time
+    ///                                 the event is fired.
+    virtual void OnAccessibilityEvent(
+        AXEventGenerator::TargetedEvent targeted_event,
+        AccessibilityBridge* bridge) = 0;
+
+    //------------------------------------------------------------------------------
+    /// @brief      Dispatch accessibility action back to the Flutter framework
+    ///
+    /// @param[in]  target              The semantics node id of the action
+    /// target.
+    /// @param[in]  action              The generated flutter semantics action.
+    /// @param[in]  data                Additional data associated with the
+    /// action.
+    /// @param[in]  data_size           The length of the additional data.
+    virtual void DispatchAccessibilityAction(uint16_t target,
```

Comment:
If we don't have one already, we should probably have a typedef somewhere for things that we treat as node IDs.

We also will need to consider whether those need to support negative numbers to signify an invalid node. 

It's not clear to me that uint16 will be large enough either - on other platforms I believe we're using 32 or 64 bit ints.